packages/opentelemetry-instrumentation-http/src/types.ts

```diff
@@ -79,6 +79,10 @@ export interface StartOutgoingSpanCustomAttributeFunction {
   (request: RequestOptions ): SpanAttributes;
 }
 
+export interface HttpCustomSpanNameFunction {
+  (componentName?: 'http' | 'https', method?: string): string;
```

Comment:
Shouldn't we include more data here into this callback? I guess the request object would be helpful.

Reply:
I think we should just give the whole `ClientRequest` object here indeed

Reply:
Agree, another option would be an object (defined with an interface) which includes the ClientRequest. This way if we need to add anything in the future it doesn't cause a break in the signature of this function -- just need to add ```optional``` extra stuff... (But that does seem a bit of an overkill for this specific case)